workflow/controller/controller.go

```diff
@@ -446,11 +496,38 @@ func (wfc *WorkflowController) processNextItem() bool {
 		return true
 	}
 
+	if wf.Spec.Synchronization != nil {
+		priority, creationTime := getWfPriority(woc.wf)
+		acquired, wfUpdate, msg, err := wfc.concurrencyMgr.TryAcquire(woc.wf, "", priority, creationTime, woc.wf.Spec.Synchronization)
+		if err != nil {
+			log.Warnf("Failed to acquire the lock for '%s' : %v", key, err)
+			woc.markWorkflowFailed(fmt.Sprintf("invalid spec: %s", err.Error()))
+			woc.persistUpdates()
+			wfc.throttler.Remove(key)
+			return true
+		}
+		woc.updated = wfUpdate
+		if !acquired {
+			log.Warnf("Workflow %s processing has been postponed due to concurrency limit. %s", key, msg)
+			woc.persistUpdates()
```

Comment:
No `wfc.throttler.Remove(key)` here? Why not?

Reply:
It needs to reconcile after 10s. if we remove the key from throttled, it will lose priority position